### ISSUE TITLE
t.w2.vhost.AutoVHostURIRewrite fails silently if no X-Forwarded-Host header is provided.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11190]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#1700|
|Type|enhancement|
|Created|2006-05-03 22:20:26Z|

Certain proxies (specifically http://www.apsis.ch/pound/) don't send X-Forwarded-Host headers instead they send the proper Host header.  t.w2.vhost.AutoVHostURIRewrite currently raises a http.HTTPError(responsecode.BADREQUEST), giving the user no indication of the problem.

<details><summary>Searchable metadata</summary>

```
trac-id__1700 1700
type__enhancement enhancement
reporter__dreid dreid
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1146694826000000 1146694826000000
changetime__1149118870000000 1149118870000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set owner to @dreid|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid commented|
|-|-|

addressed in branch [branches/AutoVHostURIRewriteImprovement-1699](branches/AutoVHostURIRewriteImprovement-1699)

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set status to `closed`|
|-|-|

(In [16979]) Mangle request.remoteAddr based on the X-Forwarded-For header, also add option to use the real Host header instead of X-Forwarded-Host for compatibility with certain proxies.  Warn the user if this compatibility isn't turned on and we didn't get an X-Forwarded-Host header.

Author: dreid
Reviewer: PenguinOfDoom
fixes [#7139](https://github.com/twisted/twisted/issues/7139),[#11190](https://github.com/twisted/twisted/issues/11190)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
